src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -2232,110 +2232,37 @@ void EmitMultiChar(RegexNode node, bool emitLengthCheck = true)
             {
                 Debug.Assert(node.Kind is RegexNodeKind.Multi, $"Unexpected type: {node.Kind}");
 
-                bool caseInsensitive = IsCaseInsensitive(node);
-
                 string str = node.Str!;
-                Debug.Assert(str.Length != 0);
+                Debug.Assert(str.Length >= 2);
 
-                const int MaxUnrollLength = 64;
-                if (str.Length <= MaxUnrollLength)
+                if (IsCaseInsensitive(node)) // StartsWith(..., XxIgnoreCase) won't necessarily be the same as char-by-char comparison
```

Comment:
I suppose this is one of the things that won't get removed after I'm done with IgnoreCase right? I say that since I don't think we'll be able to use StartsWith at that point as instead of landing in EmitMultiChar we would probably land on something that emits checks for the set of lower case and upper case

Reply:
> I suppose this is one of the things that won't get removed after I'm done with IgnoreCase right?

It should... we'll never have an IgnoreCase Multi at that point.  The only place in the tree we should still have anything IgnoreCase is on a Backreference.

Reply:
Note that in the handling of concats, we already walk through the child nodes looking for series of children that we can emit more efficiently, or at least more concisely.  We can augment that to look for a series of sets that matches the behavior you'd get from OrdinalIgnoreCase, and then emit a StartsWith("...", OrdinalIgnoreCase) for that subset _if_ Egor subsequently adds in the OrdinalIgnoreCase vectorization optimization to StartsWith.

Reply:
I do plan to add `OrdinalIgnoreCase` - it should not be difficult (not in that PR though), but should be pretty beneficial, also I found plenty of usages of StartsWith + OrdinalIgnoreCase  in aspnetcore and other popular repos